lib/task_listener.js

```diff
@@ -314,11 +319,102 @@ export default class TaskListener extends EventEmitter {
   }
 
   /**
+   * Coalesce the task in the given claim, claiming any additional tasks
+   * directly from the TC queue service, then calling runTask with the
+   * resulting task set.
+   */
+  async coalesceClaim(claim) {
+    let task = claim.task;
+    let taskId = claim.status.taskId;
+
+    try {
+      // if the task is not coalescible, then the taskset is just the one claim
+      if (!task.payload.coalescer || !task.payload.coalescer.url) {
+        return [claim];
+      }
+
+      let routes = task.routes;
+      routes = routes.filter(r => r.startsWith(COALESCER_ROUTE_PREFIX));
+      if  (routes.length != 1) {
+        return [claim];
+      }
+
+      let coalescingKey = routes[0].substr(COALESCER_ROUTE_PREFIX.length);
+      let coalescerUrl = task.payload.coalescer.url + coalescingKey;
```

Comment:
There are arguments on both sides of this: adding the slash is probably user-friendly, but really just papers over a user misconfiguration and is "magic".  The URLs will be in the task definitions, which will be in-tree, so I expect they'll not spontaneously lose their slashes.